### PR TITLE
musl exp port

### PR DESCRIPTION
**Pull request by musm**
musl exp port.

### CODE REVIEW

**Comment by ararslan** [comment]
Your enthusiasm and work here is very much appreciated, but it might be worth waiting a bit to see what the consensus is on #5. We may end up wanting to have things like this translated in an automated way from an existing C library.

**Comment by oxinabox** [comment]
> We may end up wanting to have things like this translated in an automated way from an existing C library.

Having done a bit of automated porting/code generation before,
having several chunks of code, that are manually translated is a boon for working out what your automated translator should do; and assessing if it is doing it.

**Comment by simonbyrne** [comment]
I don't think an automatic translator is likely to be forthcoming in the immediate future (though I would love to be proven wrong), so I don't see any problem with manual translations.

File: src/exp.jl [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        # underflow if x!=-inf
+        if x < -745.13321910194110842
+            return 0.0
+        end
+    end
+
+    # argument reduction
+    if hx > 0x3fd62e42 # if |x| > 0.5 ln2
+        if hx >= 0x3ff0a2b2  # if |x| >= 1.5 ln2
+            k = unsafe_trunc(Int32, invln2*x + half[sign+1])
```
Comment (vchuravy) [inline src/exp.jl]: Maybe use `ifelse(sign == 0, 0.5, -0.5)` or `0.5 - sign` instead of a const array?

Comment (musm) [inline src/exp.jl]: Yeah that's a much  better idea! I was planing on this change after I  figured out why _exp(1.0) was off in the last bits from Base.exp(1.0)

Comment (musm) [inline src/exp.jl]: I think 0.5 - sign might be faster looking at code_native it has shorter instructions

**Comment by musm** [comment]
Looks to work for everything except the following.

``` julia
julia> exp(1.0)
2.718281828459045

julia> _exp(1.0)
2.7182818284590455
```

I also need to check performance,  currently looks like 3.5 times slower, but this might be misleading.

**Pull request closed without merging**